### DISCUSSION TITLE
Migrating char(36) to binary(16) uuid primary key

### DISCUSSION THREAD

**Discussion by barbhackk** [symfony/symfony#47981] [Q&A]
Hi,

I recently migrated from **Symfony 5.2** to **Symfony** 6.

I use primary keys for entities in **guid** format stored in **CHAR (36)**. Unfortunately since the update of the version to 6, I notice that the primary key is no longer in CHAR(36) format but **BINARY(16)**. 

My tables contain data with CHAR IDs (36).

If I migrate, I risk losing data. I searched everywhere but I can't find a solution. 

Do I have to convert all identifiers into a database? 
Is it all done alone when I launch doctrine:migration:migrate?
Can we use an alternative?

My current configuration:
```
#[ORM\Id]
#[ORM\Column(type: 'uuid', unique: true)]
#[ORM\GeneratedValue(strategy: 'CUSTOM')]
#[ORM\CustomIdGenerator(class: 'doctrine.uuid_generator')]
private ?Uuid $id = null;

```
 
 Thanks for your help.

cdt

Sébastien

**Comment by MatTheCat** [comment]
Hi :wave: 

Be careful when using RC! This changed between 5.2 RC1 and RC2: https://github.com/symfony/symfony/pull/39113

> Do I have to convert all identifiers into a database?

Yes because their string and binary representation are different.

> Is it all done alone when I launch doctrine:migration:migrate?

Doctrine would generate a migration changing your columns type but it is your responsibility to convert your identifiers (using [UUID_TO_BIN](https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid-to-bin) eg.). You would typically add this conversion in the same migration.

> Can we use an alternative?

Well yes you could explicit the column’s type to `CHAR(36)` but this would not be very efficient.
